### PR TITLE
Feature: bootstrap like config

### PR DESCRIPTION
**Pull request by rgerstenberger**
This adds a default `settings` singleton for components to use. Also adds an exposed `configureSettings` method and an internal `computeSettings` method to create bootstrap equivalent computed settings properties. `configureSettings` allows for a user to pass in a hardcoded value to overwrite any computed value.  

This updates part of the buttons class to use the new settings for color but a lot more work is needed to integrate these settings.

Despite that, it's pretty cool to change one setting and see it propagate:

```
RadiumBootstrap.configureSettings({
  brandPrimary: "#fc"
});
```

results in:

screen shot 2015-02-19 at 12 41 16 pm

cc @alexlande, @aykayen @ryan-roemer

### CODE REVIEW

**Comment by alexlande** [comment]
Very cool. Nice work! One small change I'd suggest but otherwise :shipit:

**Comment by colinmegill** [comment]
+1 for generalizing as this would be useful with plain react styling
ᐧ

File: modules/components/button.jsx [hunk 1]
```diff
@@ -1,5 +1,6 @@
 var React = require('react');
 var Radium = require('radium');
+var RadiumSettings = require('../settings.js');
```
Comment (rgerstenberger) [inline modules/components/button.jsx]: I agree `RadiumSettings` is a definite misnomer.

I'm confused though, you wouldn't use this without Radium-bootstrap though, which can't be used without Radium. Are you saying this module should be available outside of Radium-bootstrap? 

What if we just call it `settings`, and you've have to use it as `RadiumBootstrap.settings` or `var { settings } = require(`RadiumBootstrap`);`

`Bootstrap settings` seems wrong to me but I'm open to suggestions, or reasons why it's right :P

Comment (alexlande) [inline modules/components/button.jsx]: `RadiumBoostrap.settings` seems fine to me. I'm not saying that it should be available outside of radium bootstrap, I'm just saying it's not related to radium.

For example, if you made a one-off inline style object that wasn't processed by radium you could still use these variables:

```
<div style={{
  background: settings.blue
}}>
```

Comment (rgerstenberger) [inline modules/components/button.jsx]: Good point! Hopefully if you're building things like that you are using settings. I'll update and merge :)

**Pull request merged**